runtime/tests/tests.rs

```diff
@@ -120,6 +121,85 @@ fn transfer_fee<E: Encode>(extrinsic: &E, fee_multiplier: Fixed64, runtime_call:
 	base_fee + fee_multiplier.saturated_multiply_accumulate(length_fee + weight_fee)
 }
 
+#[test]
+fn staking_genesis_config_works() {
+	let balance_amount = 10_000 * TransactionBaseFee::get();
+	ExtBuilder::default()
+		.initial_balance(balance_amount)
+		.build()
+		.execute_with(|| {
+			for validator in validators() {
+				let (stash, controller) = validator;
+				let staked_amount = balance_amount / 5;
+
+				// Check validator is included in currect elelcted accounts
+				assert!(Staking::current_elected().contains(&stash));
+				// Check that RewardDestination is Staked (default)
+				assert_eq!(Staking::payee(&stash), RewardDestination::Staked);				
+				// Check validator free balance
+				assert_eq!(
+					<GenericAsset as MultiCurrencyAccounting>::free_balance(&stash, Some(CENTRAPAY_ASSET_ID)),
+					balance_amount
+				);
+				// Check how much is at stake
+				assert_eq!(Staking::ledger(controller), Some(StakingLedger {
+					stash,
+					total: staked_amount,
+					active: staked_amount,
+					unlocking: vec![],
+				}));
+			}
+		});
+}
+
+#[test]
+fn staking_validators_should_received_equal_transaction_fee_reward() {
+	let transfer_amount = 50;
+	let balance_amount = 10_000 * TransactionBaseFee::get();
+	let runtime_call = Call::GenericAsset(pallet_generic_asset::Call::transfer(
+		CENTRAPAY_ASSET_ID,
+		bob(),
+		transfer_amount,
+	));
+
+	ExtBuilder::default()
+		.initial_balance(balance_amount)
+		.build()
+		.execute_with(|| {
+			let xt = sign(CheckedExtrinsic {
+				signed: Some((alice(), signed_extra(0, 0, None, None))),
+				function: runtime_call.clone(),
+			});
+	
+			let fm = TransactionPayment::next_fee_multiplier();
+			let fee = transfer_fee(&xt, fm, &runtime_call);
+			let validator_count = validators().len() as Balance;
+			let fee_reward = fee / validator_count;
+			let remainder = fee % validator_count;
+
+			let previous_total_issuance = GenericAsset::total_issuance(&CENTRAPAY_ASSET_ID);
+
+			initialize_block();
+			let r = Executive::apply_extrinsic(xt);
+			assert!(r.is_ok());
+
+			// Check total_issurance is adjusted
+			// FIXME: total_issurance is not adjusted
```

Comment:
Since we mint exactly what was burned (except remainder) total issuance won't change until we introduce an inflation parameter in the reward step

Reply:
Reward paid out now creates a positive imbalance that offsets the negative imbalance, so if there are remainders left, the overall issuance is adjusted.